component/prometheus/relabel/relabel_test.go

```diff
@@ -15,100 +17,57 @@ import (
 	"github.com/prometheus/prometheus/model/labels"
 	"github.com/prometheus/prometheus/model/relabel"
 	"github.com/prometheus/prometheus/model/value"
+	"github.com/prometheus/prometheus/storage"
 	"github.com/stretchr/testify/require"
 )
 
 func TestCache(t *testing.T) {
 	relabeller := generateRelabel(t)
-	fm := prometheus.NewFlowMetric(0, labels.FromStrings("__address__", "localhost"), 0)
-
-	relabeller.Receive(time.Now().Unix(), []*prometheus.FlowMetric{fm})
+	lbls := labels.FromStrings("__address__", "localhost")
+	relabeller.relabel(0, lbls)
 	require.Len(t, relabeller.cache, 1)
-	entry, found := relabeller.getFromCache(fm.GlobalRefID())
-	newFm := prometheus.NewFlowMetric(entry.id, entry.labels, 0)
+	entry, found := relabeller.getFromCache(prometheus.GlobalRefMapping.GetOrAddGlobalRefID(lbls))
 	require.True(t, found)
 	require.NotNil(t, entry)
-	require.True(t, newFm.GlobalRefID() != fm.GlobalRefID())
+	require.True(
+		t,
+		prometheus.GlobalRefMapping.GetOrAddGlobalRefID(entry.labels) != prometheus.GlobalRefMapping.GetOrAddGlobalRefID(lbls),
+	)
 }
 
 func TestEviction(t *testing.T) {
 	relabeller := generateRelabel(t)
-	fm := prometheus.NewFlowMetric(0, labels.FromStrings("__address__", "localhost"), 0)
-
-	relabeller.Receive(time.Now().Unix(), []*prometheus.FlowMetric{fm})
+	lbls := labels.FromStrings("__address__", "localhost")
+	relabeller.relabel(0, lbls)
 	require.Len(t, relabeller.cache, 1)
-	fmstale := prometheus.NewFlowMetric(0, labels.FromStrings("__address__", "localhost"), math.Float64frombits(value.StaleNaN))
-	relabeller.Receive(time.Now().Unix(), []*prometheus.FlowMetric{fmstale})
+	relabeller.relabel(math.Float64frombits(value.StaleNaN), lbls)
 	require.Len(t, relabeller.cache, 0)
 }
 
 func TestUpdateReset(t *testing.T) {
 	relabeller := generateRelabel(t)
-	fm := prometheus.NewFlowMetric(0, labels.FromStrings("__address__", "localhost"), 0)
-
-	relabeller.Receive(time.Now().Unix(), []*prometheus.FlowMetric{fm})
+	lbls := labels.FromStrings("__address__", "localhost")
+	relabeller.relabel(0, lbls)
 	require.Len(t, relabeller.cache, 1)
 	_ = relabeller.Update(Arguments{
 		MetricRelabelConfigs: []*flow_relabel.Config{},
 	})
 	require.Len(t, relabeller.cache, 0)
 }
 
-func TestThatValueIsNotReused(t *testing.T) {
```

Comment:
Is this test no longer necessary? It felt like a good regression test to have

Reply:
With the removal of flowmetric the cache can never be passed the value so it's not necessary. The test was mostly obsolete earlier but now its more explicit that values cannot enter the cache.